src/commands/VerifyCommand.ts

```diff
@@ -0,0 +1,116 @@
+import { Message, MessageEmbed, TextChannel } from 'discord.js';
+import PrefixCommand from './PrefixCommand';
+import BotConfig from '../BotConfig';
+import Command from './Command';
+import DiscordUtil from '../util/DiscordUtil';
+import TaskScheduler from '../tasks/TaskScheduler';
+import RemovePendingVerificationTask from '../tasks/RemovePendingVerificationTask';
+
+export default class VerifyCommand extends PrefixCommand {
+	public readonly aliases = ['verify'];
+
+	public async run( message: Message, args: string ): Promise<boolean> {
+		if ( args.length ) {
+			return false;
+		}
+
+		const pendingChannel = await DiscordUtil.getChannel( BotConfig.verification.pendingVerificationChannel );
+		const logChannel = await DiscordUtil.getChannel( BotConfig.verification.verificationLogChannel );
+
+		if ( pendingChannel instanceof TextChannel && logChannel instanceof TextChannel ) {
+
+			let foundUser = false;
+			let isVerified = false;
+
+			const allMessages = pendingChannel.messages.cache;
+			const verifications = logChannel.messages.cache;
+
+			for ( const loop of allMessages ) {
+				const thisMessage = loop[1];
+				if ( thisMessage.embeds.length == 0 ) continue;
+				if ( thisMessage.embeds[0].fields[0].value.replace( /[<>@!]/g, '' ) == message.author.id ) {
+					foundUser = true;
+					break;
+				}
+			}
+
+			for ( const loop of verifications ) {
+				const thisMessage = loop[1];
+				if ( thisMessage.embeds.length == 0 ) continue;
+				if ( thisMessage.embeds[0].fields[0].value.replace( /[<>@!]/g, '' ) == message.author.id ) {
+					isVerified = true;
+					break;
+				}
+			}
+
+			try {
+				const role = await pendingChannel.guild.roles.fetch( BotConfig.verification.verifiedRole );
+				const targetUser = await message.guild.members.fetch( message.author.id );
+
+				if ( targetUser.roles.cache.has( role.id ) || isVerified ) {
+					await message.channel.send( `${ message.author }, you have already linked your accounts!` );
+					await message.react( '❌' );
+					return false;
+				}
+			} catch ( error ) {
+				Command.logger.error( error );
+			}
+
+			if ( foundUser ) {
+				await message.channel.send( `${ message.author }, you already have a pending verification request!` );
+				await message.react( '❌' );
+				return false;
+			}
+
+		}
+
+		try {
+			const token = 'DISCORD-VERIFICATION-' + this.randomString( 15, '23456789abcdeghjkmnpqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ' );
```

Comment:
Might be good to add a comment, explaining why some chars (`0`, `o`, `O`; `1`, `I`) are omitted.
And why are `f` and `i` missing? With which characters can they be confused?

Reply:
You're definitely right. Good idea, thank you!
I don't know why I removed `f`. It's very possible that it was just a mistake.
I think I removed `i` because it could be confused with `I`, but now I realize that they look different enough in Discord that they wouldn't be confused.